application/src/main/java/org/togetherjava/tjbot/commands/free/FreeCommand.java

```diff
@@ -0,0 +1,364 @@
+        @NotNull
+        final JDA jda = event.getJDA();
+        // TODO remove this when onGuildMessageReceived has another access point
+        jda.addEventListener(this);
+
+        initChannelsToMonitor();
+        initStatusMessageChannels(jda);
+        logger.debug("Config loaded:\n{}", channelMonitor);
+
+        checkBusyStatusAllChannels(jda);
+
+        channelMonitor.statusIds()
+            .map(id -> requiresTextChannel(jda, id))
+            .forEach(this::displayStatus);
+
+        isReady = true;
+    }
+
+    /**
+     * When triggered with {@code /free} this will mark a help channel as not busy (free for another
+     * person to use).
+     * <p>
+     * If this is called on from a channel that was not configured for monitoring (see
+     * {@link FreeCommandConfig}) the user will receive an ephemeral message stating such.
+     * 
+     * @param event the event that triggered this
+     * @throws IllegalStateException if this method is called for a Global Slash Command
+     */
+    @Override
+    public void onSlashCommand(@NotNull final SlashCommandEvent event) {
+        logger.debug("/free used by {} on channel {}", event.getUser().getAsTag(),
+                event.getChannel().getName());
+        if (!handleShouldBeProcessed(event)) {
+            return;
+        }
+
+        long id = event.getChannel().getIdLong();
+        // do not need to test if key is present, shouldHandle(event) already does.
+        if (!channelMonitor.isChannelBusy(id)) {
+            FreeUtil.sendErrorMessage(event, UserStrings.ALREADY_FREE_ERROR.message());
+            return;
+        }
+        // TODO check if /free called by original author, if not put message asking if he approves
+        channelMonitor.setChannelFree(id);
+        displayStatus(channelMonitor.getStatusChannelFor(requiresGuild(event)));
+        event.reply(UserStrings.MARK_AS_FREE.message()).queue();
+    }
+
+    /**
+     * Method to test event to see if it should be processed.
+     * <p>
+     * Will respond to users describing the problem if the event should not be processed.
+     * <p>
+     * This checks if the command system is ready to process events, if the event was triggered in a
+     * monitored guild and in a monitored channel.
+     *
+     * @param event the event to test for validity.
+     * @return true if the event should be processed false otherwise.
+     */
+    private boolean handleShouldBeProcessed(@NotNull final SlashCommandEvent event) {
+        if (!isReady) {
+            logger.debug(
+                    "Slash command requested by {} in {}(channel: {}) before command is ready.",
+                    event.getUser().getIdLong(), event.getGuild(), event.getChannel().getName());
+            FreeUtil.sendErrorMessage(event, UserStrings.NOT_READY_ERROR.message());
+            return false;
+        }
+        // checks if guild is null and throws IllegalStateException if it is
+        @NotNull
```

Comment:
Same here, it feels rather pointless and inconsistent

Reply:
we are placing @NotNull on values that might be null that we are guaranteeing is not null
both for the sake of readability and to help the intellij code processing

iirc you are the person who opened the issue about using more intelliJ annotations, not less?

getGuild can return a null value in JDA
the method i have there `requiresGuild()` is not obviously preventing nulls even though it is
the @NotNull makes it obvious that this is the case.

maybe it is inconsistant, i havent rewritten my entire project
but i'm trying to include the required annotations when/where applicable

Reply:
Correct.

But I find them cluttering myself as local variables.
And IntelliJ doesn't need those annotations on local variables, it doesn't help anything.

Furthermore, I'd assume that the return value of requireGuild is NotNull, so I wouldn't use it myself either there.


I'll let the others decide to keep or remove them.
@Zabuzard

Reply:
Im not 100% decided but I dont care that much to be honest. Im fine with both at this point.

Reply:
removed